sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -456,11 +459,76 @@ public Query apply(Function input, Context context) throws IOException {
 
         class NotQuery implements FunctionToQuery {
 
+            class SymbolToNotNullContext {
+                private final ImmutableSet.Builder<Reference> references = ImmutableSet.builder();
```

Comment:
since this is mostly private internal stuff I think it would be fine to use a regular HashSet. Otherwise there's the builder instance overhead.